Cargill/splinter
# Add permission details

Adds a display name and description to all checked REST API permissions along with the `/authorization/permissions` endpoint and `splinter permissions` CLI command for viewing all permission details.

# Testing

1. Start splinterd:

```bash
SPLINTER_HOME=/tmp/splinter cargo run --manifest-path cli/Cargo.toml --features=experimental -- database migrate
SPLINTER_HOME=/tmp/splinter cargo run --manifest-path cli/Cargo.toml --features=experimental -- cert generate
SPLINTER_HOME=/tmp/splinter cargo run --manifest-path splinterd/Cargo.toml --features experimental -- --rest-api-endpoint http://localhost:8080 --tls-insecure -vv
```

2. Generate a key pair for yourself you don't already have one:

```bash
cargo run --manifest-path cli/Cargo.toml --features=experimental -- keygen
```

3. Copy your public key to the allow keys file:

```bash
cat ~/.cylinder/keys/<your-username>.pub > /tmp/splinter/data/allow_keys
```

4. List permissions using the CLI:

```bash
cargo run --manifest-path cli/Cargo.toml --features=experimental -- permissions -U http://localhost:8080
ID                              NAME                   DESCRIPTION
registry.read                   Registry read          Allows the client to read the registry
registry.write                  Registry write         Allows the client to modify the registry
circuit.read                    Circuit read           Allows the client to read circuit state
circuit.write                   Circuit write          Allows the client to modify circuit state
scabbard.write                  Scabbard write         Allows the client to submit batches to scabbard services
scabbard.read                   Scabbard read          Allows the client to read scabbard services' state and batch statuses
authorization.maintenance.read  Maintenance mode read  Allows the client to check maintenance mode status
authorization.maintenance.write Maintenance mode write Allows the client to enable/disable maintenance mode
status.read                     Status read            Allows the client to get node status info
authorization.rbac.read         RBAC read              Allows the client to read roles, identities, and role assignments
authorization.rbac.write        RBAC write             Allows the client to modify roles, identities, and role assignments
health.read                     Health read            Allows the client to check node health
authorization.permissions.read  Permissions read       Allows the client to read REST API permissions
```
